integration/clean/clean_aws_resource.go

```diff
@@ -0,0 +1,145 @@
+	//Documentation: https://docs.aws.amazon.com/cli/latest/reference/autoscaling/describe-auto-scaling-groups.html#options
+	var nextToken *string
+
+	var parameterStoreNameFilter = ssmType.ParameterStringFilter{
+		Key:    aws.String("Name"),
+		Option: aws.String("BeginsWith"),
+		Values: []string{"AmazonCloudWatch"},
+	}
+
+	for {
+		describeParametersInput := ssm.DescribeParametersInput{
+			ParameterFilters: []ssmType.ParameterStringFilter{parameterStoreNameFilter},
+			NextToken:        nextToken,
+		}
+		describeParametersOutput, err := ssmClient.DescribeParameters(ctx, &describeParametersInput)
+
+		if err != nil {
+			log.Printf("Describe Parameter Stores failed because of %v", err)
+			return
+		}
+
+		for _, parameter := range describeParametersOutput.Parameters {
+
+			if !expirationDate.After(*parameter.LastModifiedDate) {
+				continue
+			}
+
+			log.Printf("Trying to delete Parameter Store with name %s and creation date %v", *parameter.Name, *parameter.LastModifiedDate)
+
+			deleteParameterInput := ssm.DeleteParameterInput{Name: parameter.Name}
+
+			if _, err := ssmClient.DeleteParameter(ctx, &deleteParameterInput); err != nil {
+				log.Printf("Failed to delete Parameter Store with name %s because of %v", *parameter.Name, err)
+				return
+			}
+		}
+
+		if describeParametersOutput.NextToken == nil {
+			break
+		}
+
+		nextToken = describeParametersOutput.NextToken
+	}
+}
+
+func cleanAMI() {
+	ctx := context.Background()
+	defaultConfig, err := config.LoadDefaultConfig(ctx)
+
+	if err != nil {
+		log.Printf("Load default config failed because of %v", err)
+		return
+	}
+
+	ec2client := ec2.NewFromConfig(defaultConfig)
+
+	// Filter name of EC2
+	ec2NameFilter := ec2Type.Filter{Name: aws.String("name"), Values: []string{
+		"cloudwatch-agent-integration-test*",
+	}}
+
+	//get instances to delete
+	describeImagesInput := ec2.DescribeImagesInput{Filters: []ec2Type.Filter{ec2NameFilter}}
+	describeImagesOutput, err := ec2client.DescribeImages(ctx, &describeImagesInput)
+	if err != nil {
+		log.Printf("Describe images failed because of %v", err)
+		return
+	}
+
+	var errors []error
+	for _, image := range describeImagesOutput.Images {
+		creationDate, err := smithyTime.ParseDateTime(*image.CreationDate)
+		if err != nil {
+			errors = append(errors, err)
+			continue
+		}
+		log.Printf("Image name %v image id %v experation date %v creation date parsed %v image creation date raw %v",
+			*image.Name, *image.ImageId, creationDate, expirationDate, *image.CreationDate)
+		if expirationDate.After(creationDate) {
+			log.Printf("Try to delete ami %s tags %v launch-date %s", *image.Name, image.Tags, *image.CreationDate)
+			deregisterImageInput := ec2.DeregisterImageInput{ImageId: image.ImageId}
+			_, err := ec2client.DeregisterImage(ctx, &deregisterImageInput)
+			if err != nil {
+				errors = append(errors, err)
+			}
+		}
+	}
+
+	if len(errors) != 0 {
+		log.Printf("Deleted some of AMIs failed because of %v", err)
+		return
```

Comment:
This return statement doesn't add value. Whether the errors array is empty or not, we still return after this anyway.

Reply:
I added the `log.Printf` after this `len(errors)` and changes back to `log.Fatalf` when the cleaning encouters errors.